routers/web/auth/auth.go

```diff
@@ -160,54 +160,42 @@ func CheckAutoLogin(ctx *context.Context) bool {
 	return false
 }
 
-// SignIn render sign in page
-func SignIn(ctx *context.Context) {
+func prepareSignPageData(ctx *context.Context) {
 	ctx.Data["Title"] = ctx.Tr("sign_in")
-
-	if CheckAutoLogin(ctx) {
-		return
-	}
-
-	if ctx.IsSigned {
-		RedirectAfterLogin(ctx)
-		return
-	}
-
-	oauth2Providers, err := oauth2.GetOAuth2Providers(ctx, optional.Some(true))
-	if err != nil {
-		ctx.ServerError("UserSignIn", err)
-		return
-	}
-	ctx.Data["OAuth2Providers"] = oauth2Providers
+	ctx.Data["OAuth2Providers"], _ = oauth2.GetOAuth2Providers(ctx, optional.Some(true))
 	ctx.Data["Title"] = ctx.Tr("sign_in")
 	ctx.Data["SignInLink"] = setting.AppSubURL + "/user/login"
 	ctx.Data["PageIsSignIn"] = true
 	ctx.Data["PageIsLogin"] = true
 	ctx.Data["EnableSSPI"] = auth.IsSSPIEnabled(ctx)
+	ctx.Data["EnablePasswordLoginForm"] = setting.Service.EnablePasswordLoginForm
 
 	if setting.Service.EnableCaptcha && setting.Service.RequireCaptchaForLogin {
 		context.SetCaptchaData(ctx)
 	}
+}
 
+// SignIn render sign in page
+func SignIn(ctx *context.Context) {
+	if CheckAutoLogin(ctx) {
+		return
+	}
+	if ctx.IsSigned {
+		RedirectAfterLogin(ctx)
+		return
+	}
+	prepareSignPageData(ctx)
 	ctx.HTML(http.StatusOK, tplSignIn)
 }
 
 // SignInPost response for sign in request
 func SignInPost(ctx *context.Context) {
-	ctx.Data["Title"] = ctx.Tr("sign_in")
-
-	oauth2Providers, err := oauth2.GetOAuth2Providers(ctx, optional.Some(true))
-	if err != nil {
-		ctx.ServerError("UserSignIn", err)
+	if !setting.Service.EnablePasswordLoginForm {
```

Comment:
By the way, doesn't this imply that `DISABLE_REGISTRATION` is `true` as well?
Or what's the use of offering a sign-up page where you can set a password that you cannot use afterward?
Perhaps we should add a warning in the logs when `!setting.Service.EnablePasswordLoginForm && DisableRegistration`.

Reply:
No, that's unrelated.

An account could be created manually with DISABLE_REGISTRATION=true, and then make it login by password

Reply:
I think we are talking about different points:

>  and then make it login by password

How if the password login is disabled?

I am not talking about the case `registration is disabled and thus login should be disabled`, I'm talking about the other way around:
`password login is disabled thus (password) registration should be disabled`

Reply:
Hmm, you reminded me that we should use the same wording ....

renamed all names from "login" to "signin" in bb7f05424aa55e73ded56ec99406429eeabe83d7

Reply:
> How if the password login is disabled?

But ENABLE_BASIC_AUTHENTICATION could still be true.

Reply:
Wait, now I'm confused:
What's the difference between `ENABLE_PASSWORD_SIGNIN_FORM` and `ENABLE_BASIC_AUTHENTICATION`?
**Where** can you still use your password?
For the API (where IIRC it is deprecated for removal) or `git push`?
Or somewhere else?

Reply:
> For the API (where IIRC it is deprecated for removal) or `git push`?

yes, including API